lib/index.js

```diff
@@ -84,8 +83,8 @@ api.add = callbackify(async (meta = {}, options = {}) => {
     }
   }
 
-  // generate UUIDs for the ledger storage
-  const ledgerUuid = uuid();
+  // remove `urn:uuid:` from the ledgerNodeId
+  const ledgerUuid = options.ledgerNodeId.substr(9);
```

Comment:
This conflates the ledger ID with the ledgerNode ID ... which is a problem and we may already be confusing people with it in other places. The ledger ID is the same across many different ledger nodes (ledger nodes that all mirror the same ledger).

I'm ok with more strongly tying a particular storage instance to a particular ledger node ... but don't want to do it this way. So, previously (as you can see in this file) we were tying storage to a particular ledger. That seems like a mistake. Instead, it *should* be tied to a ledger node like you're doing in this patch. So if we can leave the `ledgerUuid` alone and instead use `ledgerNodeId` as the prefix for the collection names that would be ideal.

I'm also a little apprehensive about the blind stripping of `urn:uuid:` from `ledgerNodeId` because it assumes that format. Seems like a bad idea. Do we need to do that or can we just leave the ledgerNodeId alone? (Is `:` ok in a collection name?)

Reply:
I know it's not very clear in the code, but the `ledger ID` is passed into this API and that is used.  That the `did:v1:` style ID.  I'll clean up the code a bit to make that more clear.

I did test using the entire `ledgerNodeId` as the collection prefix.  The undesirable side-effect of that is that in a number of cases (just in this module) the index names extend beyond 128 chars and mongo throws an error.  The solution to this is to use named indexes in those cases.  However, unless we *always and everywhere* use named indexes and `ledgerNodeId` becomes some arbitrarily long URI, we can expect to have problems in the future.

Yes, a `:` is acceptable in the collection name.  But I've already mentioned other difficulties related to using the full `ledgerNodeId`.

Reply:
@dlongley do you know why the `ledger ID` (did:v1:) is part of this API in the first place?  That information is recorded in the `ledgerNode` collection like:
```
{
    "_id": ObjectID("5b7f0a819b478a068e2636fc"),
    "id": "W9jE3L7XnEQqwf7vGmL8cSqVXlN8EGdG83WVOO5dU+8=",
    "ledger": "4rTScOEGjUe8CirBvF6yEYfCZCnRU4ukNSA+MgPPDE0=",
    "storage": "GfliNEgNspsKTAXr0XsTBxOJwvr1B0ohi8HKiKL9ojo=",
    "owner": "5eSA5xK80hNnoeBs+ZipbUBWa6TF9cxVeFOd9J7EzRM=",
    "ledgerNode": {
        "id": "urn:uuid:a12ec6e4-2388-441b-b057-f194d8cd8e80",
        "ledger": "did:ledgertest:eb8c22dc-bde6-4315-92e2-59bd3f3c7d59",
        "owner": "did:7e4a0145-c821-4e56-b41e-2e73e1b0615f",
        "peerLedgerAgent": [],
        "storage": {
            "plugin": "mongodb",
            "id": "urn:uuid:d9aec16d-3f68-4d90-9722-cdd5266ab796"
        }
    },
    "meta": {
        "created": 1535052416708,
        "updated": 1535126195760,
        "workSession": {
            "id": "906d29d7-dc0f-4c70-96e1-fb631bb39196",
            "expires": 1535126555760
        }
    }
}
```

Reply:
@mattcollier... no, I don't know why, probably just early design madness. If we have it recorded with the ledger node and the storage is bound to the ledger node that should be sufficient. I would hope anyway. I don't think we want it stored in multiple places, that just results in inconsistency bugs.

Regarding the ledgerNodeId ... I get the problem. I understand how it also doesn't solve this particular issue if we were to hash it as that wouldn't really help with debugging. We could put limits on the ledgerNodeId length ... but we may *actually* want to use named indexes everywhere because it sounds like we have a bug waiting to happen.

I'm afraid that the ledger node ID stuff is probably at the Web Ledger layer ... which means it can be any URI. That makes me think we may need to go the hash route w/storage and come up with some other easy debugging mechanism.